c2corg_ui/static/js/langservice.js

```diff
@@ -0,0 +1,121 @@
+goog.provide('app.Lang');
+
+goog.require('app');
+goog.require('ngeo.GetBrowserLanguage');
+
+
+/**
+ * @param {angular.$cookies} $cookies Cookies service.
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
+ * @param {ngeo.GetBrowserLanguage} ngeoGetBrowserLanguage
+ *        GetBrowserLanguage Service.
+ * @param {Array.<string>} langs List of available langs.
+ * @param {amMoment} amMoment angular moment directive.
+ * @param {app.Api} appApi Api service.
+ * @param {app.Authentication} appAuthentication Authentication service.
+ * @param {string} langUrlTemplate Language URL template.
+ * @constructor
+ * @ngInject
+ * @struct
+ */
+app.Lang = function($cookies, gettextCatalog, ngeoGetBrowserLanguage, langs,
+  amMoment, appApi, appAuthentication, langUrlTemplate) {
+
+  /**
+   * @type {angular.$cookies}
+   * @private
+   */
+  this.cookies_ = $cookies;
+
+  /**
+   * @type {angularGettext.Catalog}
+   * @private
+   */
+  this.gettextCatalog_ = gettextCatalog;
+
+  /**
+   * @type {ngeo.GetBrowserLanguage}
+   * @private
+   */
+  this.ngeoGetBrowserLanguage_ = ngeoGetBrowserLanguage;
+
+  /**
+   * @type {Array.<string>}
+   * @private
+   */
+  this.langs_ = langs;
+
+  /**
+   * @type {amMoment}
+   */
+  this.amMoment_ = amMoment;
+
+  /**
+   * @type {app.Api}
+   * @private
+   */
+  this.api_ = appApi;
+
+  /**
+   * @type {app.Authentication}
+   * @private
+   */
+  this.appAuthentication_ = appAuthentication;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.langUrlTemplate_ = langUrlTemplate;
+};
+
+
+/**
+ * @return {Array.<string>}
+ */
+app.Lang.prototype.getAvailableLangs = function() {
+  return this.langs_;
+};
+
+
+/**
+ * @return {string}
+ */
+app.Lang.prototype.detectLang = function() {
+  return this.cookies_.get('interface_lang') ||
+    this.ngeoGetBrowserLanguage_(this.langs_) || 'fr';
```

Comment:
Can you do it like in the previous `LangController`: In the constructor get the language (from the cookie/browser/default) and set it on `gettextCatalog` as current language. Then if you need to know the current language you call `langService.getLang()` which returns `gettextCatalog.getCurrentLanguage()`.